gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/http/HttpExecutor.java

```diff
@@ -166,7 +164,9 @@ private void setError(Throwable error) {
         response.setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
         response.setStatusMessage(HttpResponseStatus.INTERNAL_SERVER_ERROR.reasonPhrase());
         response.headers().add("X-Exception", String.valueOf(error.getMessage())); //$NON-NLS-1$
-        response.write(ExceptionUtils.getStackTrace(error));
+        response.headers().add(HttpHeaders.CONTENT_TYPE,  MediaType.APPLICATION_JSON);
+        response.write(String.format("{\"responseCode\":%d, \"message\":\"%s\"}",
+                HttpResponseStatus.INTERNAL_SERVER_ERROR.code(), error.getMessage()));
```

Comment:
I think you need to use jackson here - this code will produce an invalid response payload if the message contains quotes.  It's also missing the stack trace, which is supposed to be part of the response (but isn't tested due to the inconsistency of stack traces).  See here:

https://github.com/apiman/apiman/blob/master/gateway/platforms/servlet/src/main/java/io/apiman/gateway/platforms/servlet/GatewayServlet.java#L373-L389